aten/src/ATen/native/CPUBlas.cpp

```diff
@@ -322,9 +322,34 @@ void gemm(
      return;
    }
 #endif
+   std::vector<float> float_c(n * ldc);
+   std::vector<float> float_a(k * lda);
+   std::vector<float> float_b(n * ldb);
+   for (auto& av : float_a) {
```

Comment:
When tensor a or b is not contiguous, that is to say, lda != m or lab != k, can we still use `a++` or `b++` here ?

Reply:
To do this, you probably want to do `a.to(at::kFloat, at::MemoryFormat::Contiguous)`, which you know for sure it's properly vectorized. 

Even then, doesn't `gemm_stub` dispatch to `gemm_core_`, which already does this via `opmath` directly in the kernel?